apps/pkeyutl.c

```diff
@@ -419,8 +419,13 @@ static EVP_PKEY_CTX *init_ctx(int *pkeysize,
     if (!pkey)
         goto end;
 
-    ctx = EVP_PKEY_CTX_new(pkey, e);
-
+    if ((keyform == FORMAT_ENGINE) && (strncmp(ENGINE_get_name(e),"pkcs11 engine", strlen("pkcs11 engine"))==0)) {
+      fprintf(stderr, "engine name = \"%s\"\n", ENGINE_get_name(e));
+      ctx = EVP_PKEY_CTX_new(pkey, NULL);
+    } else {
+      ctx = EVP_PKEY_CTX_new(pkey, e);
+    }
```

Comment:
You are reading your code wrong, because change was suggested by Dr. Henson in mid-December in response to my complaint on the openssl-dev mailing list. I have tested this change with two PIV tokens, and it proved to make the difference between pkeyutl successfully using the key on the token and reporting inability to access the key. Please check with Dr. Brendon, or test for yourself - a PIV token would only set your account  back by $40.